henrietta_chow/week01/mta/js/main.js

```diff
@@ -0,0 +1,101 @@
+var mta = {
+  N: ['Times Square', '34th', '28th', '23rd', 'Union Square', '8th'],
+  L: ['8th', '6th', 'Union Square', '3rd', '1st'],
+  6: ['Grand Central', '33rd', '28th', '23rd', 'Union Square', 'Astor Place'],
+
+  getStationIndex: function(line, station) {
+    return this[line].indexOf( station ); //instead of running a loop to return index
+  },
+
+  planTrip: function(startLine, startStation, endLine, endStation) {
+    var startIndex = this.getStationIndex(startLine, startStation);
+    var endIndex = this.getStationIndex(endLine, endStation);
+    var startLineUnionSqIndex = this.getStationIndex(startLine, 'Union Square');
+    var endLineUnionSqIndex = this.getStationIndex(endLine, 'Union Square');
+
+    var sameLineTrip = function (startIndex, endIndex) {
+      var result = mta[startLine].slice((startIndex + 1), (endIndex + 1))
+      return result;
+    };
+
+    var reverseTrip = function (startIndex, endIndex) {
+      var result = (mta[startLine].slice(endIndex, startIndex)).reverse()
```

Comment:
Watch out here - `startLine` is not passed into this function as an argument, so it's not clear where it's coming from.... must be a global variable. Can you be sure it will have the value you expect in it? Same goes for `endLine` below.